web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -509,6 +510,162 @@ const EncryptionPasswordField = ({
   );
 };
 
+/**
+ * Form for configuring the space policy.
+ * @component
+ *
+ * @param {object} props
+ * @param {string} props.id - Form ID.
+ * @param {ProposalSettings} props.settings - Settings used for calculating a proposal.
+ * @param {onSubmitFn} [props.onSubmit=noop] - On submit callback.
+ *
+ * @callback onSubmitFn
+ * @param {string} policy - Name of the selected policy.
+ */
+const SpacePolicySettingsForm = ({
+  id,
+  settings,
+  onSubmit: onSubmitProp = noop
+}) => {
+  const [spacePolicy, setSpacePolicy] = useState(settings);
+
+  const selectDelete = () => {
+    setSpacePolicy("delete");
+  };
+
+  const selectResize = () => {
+    setSpacePolicy("resize");
+  };
+
+  const selectKeep = () => {
+    setSpacePolicy("keep");
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    onSubmitProp(spacePolicy);
+  };
+
+  return (
+    <Form id={id} onSubmit={onSubmit}>
+      <div className="split">
+        <span>{_("Policy to make enough free space")}</span>
+        <ToggleGroup isCompact>
+          <ToggleGroupItem
+            text={_("Delete Anything")}
+            buttonId="deletePolicy"
+            isSelected={spacePolicy === "delete"}
+            onClick={selectDelete}
+          />
+          <ToggleGroupItem
+            text={_("Resize Only")}
+            buttonId="resizePolicy"
+            isSelected={spacePolicy === "resize"}
+            onClick={selectResize}
+          />
+          <ToggleGroupItem
+            text={_("Keep Existing")}
```

Comment:
```suggestion
            text={_("Keep Everything")}
```

Help text

> Existing partitions and their data* will not be touched. There needs to be enough unpartitioned space on the target disk already.
>
> *: If there is an EFI boot partition, its contents will be modified to allow booting the system being installed.

am I correct in this?